src/NodeVisitor/LoopNodeVisitor.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\NodeVisitor;
+
+use Twig\Environment;
+use Twig\Error\SyntaxError;
+use Twig\Node\BlockNode;
+use Twig\Node\BreakNode;
+use Twig\Node\ForNode;
+use Twig\Node\MacroNode;
+use Twig\Node\Node;
+use Twig\Node\WithNode;
+
+final class LoopNodeVisitor implements NodeVisitorInterface
+{
+    /**
+     * @var int
+     */
+    private $depth = 0;
+
+    /**
+     * @var int[]
+     */
+    private $structureStack = [];
+
+    /**
+     * @var Node|null
+     */
+    private $elseNode;
+
+    public function enterNode(Node $node, Environment $env): Node
+    {
+        if ($this->isStructureNode($node)) {
+            $this->structureStack[] = [$node, $this->depth, $this->elseNode];
+            $this->depth = 0;
+        } elseif ($node instanceof ForNode) {
+            $this->elseNode = $node->hasNode('else') ? $node->getNode('else') : null;
+            ++$this->depth;
+        } elseif ($node instanceof BreakNode) {
+            $target = $node->getAttribute('target');
+
+            if (!\is_int($target) || $target < 1) {
+                throw new SyntaxError(sprintf('Break target must be an integer > 1, got "%s".', var_export($target, true)), $node->getTemplateLine(), $node->getSourceContext());
```

Comment:
I would avoid using `var_export()` here. What about having a different error for when this is an invalid integer and when it's not an integer? For the second case, we could use what we are using everywhere else: `\is_object($target) ? \get_class($target) : \gettype($target)`

Reply:
run into a recursive `var_export` the other day... which is not supported.., good idea to avoid, updated, thanks!